api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -41,12 +48,19 @@
       userRecentResource -> {
         RecentResource resource = new RecentResource();
         resource.setCohort(TO_CLIENT_COHORT.apply(userRecentResource.getCohort()));
-        if (userRecentResource.getNotebookName() != null) {
-          FileDetail fileDetail = new FileDetail();
-          String[] notebookDetails = userRecentResource.getNotebookName().split(NOTEBOOKS_WORKSPACE_DIRECTORY);
-          fileDetail.setPath(notebookDetails[0] + NOTEBOOKS_WORKSPACE_DIRECTORY);
-          fileDetail.setName(notebookDetails[1]);
-          resource.setNotebook(fileDetail);
+        String notebookName = userRecentResource.getNotebookName();
+        if (notebookName != null && !notebookName.isEmpty()) {
+          try {
+            URI notebookUri = new URI(notebookName);
+            Path path = Paths.get(notebookUri.getPath());
+            String fileName = path.getFileName().toString();
+            String filePath = notebookName.replace(fileName, "");
+            FileDetail fileDetail = new FileDetail().name(fileName).path(filePath);
+            resource.setNotebook(fileDetail);
+          } catch (InvalidPathException | URISyntaxException e) {
+            log.log(Level.WARNING,
```

Comment:
I would say this is severe, i.e. bad data in the DB. Debatable whether we'd want to return the resource in this state as it would likely cause errors in the UI - I might prefer to just throw the ServerError (or log severe and don't return this one, but that might complicate the code beyond a reasonable level).

Reply:
Severe also works for me ... but I would not like to throw a server error to the user if we did something wrong when storing the path information. If it's bad, I think it best to log it so we can deal with it and move along because the user won't be able to do anything about it and even if it were returned, wouldn't be able to do anything with it.

Reply:
Done